### PR TITLE
Added glog in place of log.*-and-fmt.Error* for cmd/minikube,

### PR DESCRIPTION
**Pull request by aaron-prindle**
cmd/localkube, pkg/minikube and pkg/localkube
#56 

I was unable to get one line changed to use glog due to a "glog" importing issue I was having. I could not import "glog" into the file cmd/localkube/main.go. I still need to test the logging with localkube and write tests but the startcommand for localkube should have the correct flags, I'm just not sure what the default directory is for glog

### CODE REVIEW

File: pkg/minikube/constants/constants.go [hunk 1]
```diff
@@ -39,4 +39,14 @@ func MakeMiniPath(fileName string) string {
 	return filepath.Join(Minipath, fileName)
 }
 
+var LogFlags = [...]string{
```
Comment (dlorenc) [inline pkg/minikube/constants/constants.go]: What dp you think about making this a struct instead. Something like:

type GlogFlags struct {
  LogToStderr bool
  AlsoLogToStderr bool
...
}


Then you can define a ToFlags method that turns it into a string of flag values.

Then the GetStartCommand can take that config struct, call .ToFlags() and use a simple format string.

**Comment by aaron-prindle** [comment]
I think this is a good idea and will prevent the issues I had had with testing the code.  I will change the code to use a struct like the one above, thanks!

File: pkg/minikube/cluster/commands.go [hunk 2]
```diff
@@ -16,10 +16,27 @@ limitations under the License.
...
+	"sort"
+)
 
 // Kill any running instances.
 var stopCommand = "killall localkube | true"
+
+func GetStartCommandWithFlags(flagMap map[string]string) string {
+	startCommandFormattable := `
+# Run with nohup so it stays up. Redirect logs to useful places.
+PATH=/usr/local/sbin:$PATH nohup sudo /usr/local/bin/localkube start %s--generate-certs=false > /var/log/localkube.out 2> /var/log/localkube.err < /dev/null &
```
Comment (luxas) [inline pkg/minikube/cluster/commands.go]: Replace `/var/log/localkube.{out,err}` with arguments too, and replace them with the constants in your other PR after you rebase

File: pkg/minikube/constants/constants.go [hunk 3]
```diff
@@ -39,4 +39,14 @@ func MakeMiniPath(fileName string) string {
...
+	"logtostderr",
+	"alsologtostderr",
+	"stderrthreshold",
+	"logdir",
+	"logbacktraceat",
+	"v",
+	"vmodule",
```
Comment (luxas) [inline pkg/minikube/constants/constants.go]: Are all these flags required to register on both minikube and localkube, or are they for internal use?

Comment (aaron-prindle) [inline pkg/minikube/constants/constants.go]: These are optional flags for configuring glog and are intended only for internal use

**Comment by luxas** [comment]
LGTM, the struct would be an slight improvement, but it's okay for now

I'm letting @dlorenc make the merge.

File: pkg/minikube/cluster/commands.go [hunk 4]
```diff
@@ -16,10 +16,33 @@ limitations under the License.
...
+
+func GetStartCommand() string {
+	startCommandFormattable := `
+# Run with nohup so it stays up. Redirect logs to useful places.
+PATH=/usr/local/sbin:$PATH nohup sudo /usr/local/bin/localkube start %s--generate-certs=false > %s 2> %s < /dev/null &
+`
+	concatFlags := ""
+	for _, logFlag := range constants.LogFlags {
+		if logVal := gflag.Lookup(logFlag); logVal != nil {
+			if logVal.Value.String() != "" {
```
Comment (dlorenc) [inline pkg/minikube/cluster/commands.go]: Nit:
You could do something like this:
flagVals := []string
if val := logVal.Value.String(); val != "" {
  flagVals = append(flagVals, fmt.Springtf("--%s %s", logFlag, val))

flags := strings.Join(flagVals, " ")

File: pkg/minikube/cluster/commands.go [hunk 5]
```diff
@@ -16,10 +16,33 @@ limitations under the License.
...
+
+const (
+	remoteLocalKubeErrPath = "/var/log/localkube.err"
+	remoteLocalKubeOutPath = "/var/log/localkube.out"
+)
 
 // Kill any running instances.
 var stopCommand = "killall localkube | true"
+
+func GetStartCommand() string {
```
Comment (dlorenc) [inline pkg/minikube/cluster/commands.go]: Also could you add a few unit tests for this function? You might have to change this function so the flags are passed in instead of getting them from the global flag package.

**Comment by luxas** [comment]
This needs a rebase again, because your other logging-pr was merged...

File: pkg/localkube/proxy.go [hunk 6]
```diff
@@ -39,7 +39,7 @@ func StartProxyServer(lk LocalkubeServer) func() error {
 	config.Master = lk.GetAPIServerInsecureURL()
 
 	// TODO: investigate why IP tables is not working
-	config.Mode = componentconfig.ProxyModeUserspace
+	config.Mode = componentconfig.ProxyModeIPTables
```
Comment (dlorenc) [inline pkg/localkube/proxy.go]: I think you meant to put this change in a different commit :)

File: pkg/minikube/cluster/commands_test.go [hunk 7]
```diff
@@ -0,0 +1,76 @@
...
+	}
+}
+
+func flagMapToSetFlags(flagMap map[string]string) {
+	for k, v := range flagMap {
+		gflag.Set(k, v)
+	}
+}
+
+func genStartCommandFromFlagMap(flagMap map[string]string) string {
```
Comment (dlorenc) [inline pkg/minikube/cluster/commands_test.go]: This function looks basically the same as the implementation above, which means the test will be a little brittle. What about just asserting that each flag and it's value is present in the list (since we don't necessarily care about order). Something like

getSingleFlagValue(flag, value string) string {
  return fmt.Sprintf("--%s %s", flag, value)
}

Then your test loop over everyting in flagMap and assert that the individual flag is present with strings.Contains

**Comment by dlorenc** [comment]
LGTM.
